core/src/main/scala/kafka/coordinator/group/CoordinatorPartitionWriter.scala

```diff
@@ -165,4 +166,21 @@ class CoordinatorPartitionWriter(
     // Required offset.
     partitionResult.lastOffset + 1
   }
+
+  override def deleteRecords(tp: TopicPartition, deleteBeforeOffset: Long): Unit = {
+    var deleteResults: Map[TopicPartition, DeleteRecordsPartitionResult] = Map.empty
+    replicaManager.deleteRecords(
+      timeout = 0L,
```

Comment:
I have doubts about using timeout=0 here. My understanding is that the it will delete records from the local logs but it wont wait on the replication to acknowledges the deletion from the replicas. Hence, I suppose that it will always throw an error. I suppose that we don't really care about the replication in this case but it may spam the logs with unnecessary errors.

Reply:
This is the same as append records https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/coordinator/group/CoordinatorPartitionWriter.scala#L144

Reply:
You cannot compare with append records in this case because they don't work the sam way. In append records, the key is that we use `requiredAcks = 1` which means that the operation does not go to the purgatory at all. In the delete record case, it goes to the purgatory in all cases.

With the timeout, I believe that the following code will actually throw a `IllegalStateException` because the operation in the purgatory won't be completed when you reach it.

```
    val partitionResult = deleteResults.getOrElse(tp,
      throw new IllegalStateException(s"Delete status $deleteResults should have partition $tp."))
```

Could you please double check?

One option may be to make delete records return a future and complete it when the callback returns. This would also allow you to not trigger the next purge while the current one is not completed.